### PR TITLE
Catch errors sooner so users can recover more easily

### PR DESCRIPTION
**Pull request by foldleft**
Now with event type!

This change means two things:
* If an error happens in an EventTile, we render the EventTile as this error instead of bailing out of the entire app
* If an error happens above the EventTile level, but below the room level, we render the error for the room rather than bailing out of the entire app

In this way the user can make progress where previously we would have just hard-crashed.

Screenshot 2020-02-25 at 10 33 18

### CODE REVIEW

File: res/css/views/rooms/_EventTile.scss [hunk 1]
```diff
@@ -653,3 +653,17 @@ div.mx_EventTile_notSent.mx_EventTile_redacted .mx_UnknownBody {
...
+    .mx_EventTile_line span {
+        padding: 4px 8px;
+        border-radius: 11px;
+        box-shadow: 0px 0px 3px red inset;
+    }
+
+    a {
+        margin-left: 1em;
+    }
+}
```
Comment (turt2live) [inline res/css/views/rooms/_EventTile.scss]: newline

Comment (foldleft) [reply res/css/views/rooms/_EventTile.scss]: Fixed, and fixed the underlying issue too

Screenshot 2020-02-25 at 10 21 01

**Comment by foldleft** [comment]
is a design review a blocker?

**Comment by turt2live** [comment]
yes, as far as I know we're not supposed to be doing freeform design which affects non-devs. Rolling our own design behind labs or in devtools is largely fine (though the team should still be consulted), but something as critical as the app exploding needs to ensure the messaging, design, and feel of the app isn't going to drive people away.

**Comment by nadonomy** [comment]
@foldleft can you clarify what types of messages an EventTile could be? Is it all?

**Comment by foldleft** [comment]
Yes.  Any `EventTile` that fails to render will be rendered like this instead.  This shouldn't appear for unknown event types because no tiles are rendered for those, as I understand it.  Even so, the simpler a handler is, the less likely it is for the user to see this.

**Comment by nadonomy** [comment]
@foldleft apologies this dropped off. After looking in detail would you mind:

- Removing the stroke/box shadow
- Ensuring it's horizontally centred— looks like it's not from the screenshot but also hard to tell from the way its cropped
- Rewording "An error occurred while rendering this event. [$error]" to "Can't load this message ($error)."
- Rewording "Submit debug logs" to "Submit logs"

Thanks!

**Comment by foldleft** [comment]
How's this?

Screenshot 2020-03-26 at 13 46 37

**Comment by foldleft** [comment]
Here it is again, with css fixed, at three widths:

Screenshot 2020-04-06 at 11 45 21
Screenshot 2020-04-06 at 11 45 32
Screenshot 2020-04-06 at 11 45 42

**Pull request merged**